src/System.Linq/tests/SelectTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class SelectTests
+    {
+        [Fact]
+        public void Select_SourceIsNull_ArgumentNullExceptionThrown()
+        {
+            IEnumerable<int> source = null;
+            Func<int, int> selector = i => i + 1;
+
+            AssertException<ArgumentNullException>(() => source.Select(i => i + 1));
```

Comment:
Getting following error when trying to use it: 

> SelectTests.cs(15,13): error CS0012: The type '`Task`' is defined in an assembly
that is not referenced. You must add a reference to assembly 'System.Threading.
Tasks, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [d:\
GitHub\corefx\src\System.Linq\tests\System.Linq.Tests.csproj]

I didn't want to do that, but if you say it's ok, I'm fine with that.